reader-shp/src/main/java/com/graphhopper/reader/shp/GraphhopperShp.java

```diff
@@ -0,0 +1,26 @@
+package com.graphhopper.reader.shp;
+
+import java.util.HashSet;
+
+import com.graphhopper.GraphHopper;
+import com.graphhopper.reader.DataReader;
+import com.graphhopper.reader.shp.OSMShapeFileReader.EdgeAddedListener;
+import com.graphhopper.storage.GraphHopperStorage;
+
+public class GraphhopperShp extends GraphHopper {
+	private final HashSet<EdgeAddedListener> edgeAddedListeners = new HashSet<>();
+
+	@Override
+	protected DataReader createReader(GraphHopperStorage ghStorage) {
+		OSMShapeFileReader reader = new OSMShapeFileReader(ghStorage);
+		for (EdgeAddedListener l : edgeAddedListeners) {
+			reader.addListener(l);
+		}
+		return initDataReader(reader);
+	}
+
+	public void addListener(EdgeAddedListener l) {
```

Comment:
The listeners don't modify anything, they're only used to help me output debugging information when checking over the building of the graph. They're not actually called from the commited code (just an external main I was using to run a debug from). They are useful though, so I'd prefer keep them in temporarily.